chord_metadata_service/chord/export_cbio.py

```diff
@@ -0,0 +1,277 @@
+import logging
+import csv
+from typing import TextIO, Callable
+
+from .export_utils import ExportError
+
+from chord_metadata_service.chord.models import Dataset
+from chord_metadata_service.patients.models import Individual
+from chord_metadata_service.phenopackets import models as pm
+
+__all__ = [
+    "study_export",
+]
+
+logger = logging.getLogger(__name__)
+
+# predefined filenames recognized by cBioPortal
+STUDY_FILENAME = "meta_study.txt"
+SAMPLE_DATA_FILENAME = "data_clinical_sample.txt"
+SAMPLE_META_FILENAME = "meta_clinical_sample.txt"
+PATIENT_DATA_FILENAME = "data_clinical_patient.txt"
+PATIENT_META_FILENAME = "meta_clinical_patient.txt"
+
+CBIO_FILES_SET = frozenset({
+    STUDY_FILENAME,
+    SAMPLE_DATA_FILENAME,
+    SAMPLE_META_FILENAME,
+    PATIENT_DATA_FILENAME,
+    PATIENT_META_FILENAME
+})
+
+PATIENT_DATATYPE = 'PATIENT'
+SAMPLE_DATATYPE = 'SAMPLE'
+
+
+def study_export(getPath: Callable[[str], str], dataset_id: str):
+    """Export a given Project as a cBioPortal study"""
+    # TODO: a Dataset is a Study (associated with a publication), not a Project!
+    if Dataset.objects.count == 0:
+        raise ExportError("No Dataset to export")
+    dataset = Dataset.objects.get(identifier=dataset_id)
+    cbio_study_id = str(dataset.identifier)
+
+    # Export study file
+    with open(getPath(STUDY_FILENAME), 'w') as file_study:
+        study_export_meta(dataset, file_study)
+
+    # Export patients.
+    with open(getPath(PATIENT_DATA_FILENAME), 'w') as file_patient:
+        # Note: plural in `phenopackets` is intentional (related_name property in model)
+        indiv = Individual.objects.filter(phenopackets__table__ownership_record__dataset_id=dataset.identifier)
+        individual_export(indiv, file_patient)
+
+    with open(getPath(PATIENT_META_FILENAME), 'w') as file_patient_meta:
+        clinical_meta_export(cbio_study_id, PATIENT_DATATYPE, file_patient_meta)
+
+    # Export samples
+    with open(getPath(SAMPLE_DATA_FILENAME), 'w') as file_sample:
+        sampl = pm.Biosample.objects.filter(phenopacket__table__ownership_record__dataset_id=dataset.identifier)
```

Comment:
maybe here you can use [prefetch_related()](https://docs.djangoproject.com/en/4.0/ref/models/querysets/#prefetch-related) and add information about patient id in the final queryset that is later passed to `sample_export()`

Reply:
Thanks for the pointer to `prefetch_related`. What I did is adding a test inside a loop over the biosamples.
I am not sure about which one is better: Django documentation states that prefetch_related implies that the whole content of the Biosample table will have to be dumped in memory so that the "related" queries can be prefetched. On the other hand, querying the DB in a loop (albeit conditionally in that case) is not a great option for performance.
Maybe this could be addressed later, when we want to export more data from the biosample: prefetch_related might become unavoidable then.